dojo/jira_link/helper.py

```diff
@@ -807,6 +814,14 @@ def update_epic(engagement):
 @task
 @dojo_model_from_id(model=Engagement)
 def add_epic(engagement):
+    add_epic_sync(engagement)
+
+
+# This code was not being tested, but calling it
+# as a task was not working. Using block execution
+# would not either here because this method is called
+# explicitly instead of through a save function
+def add_epic_sync(engagement):
```

Comment:
what was the error seen when doing async? I think we should keep everything related to JIRA in  celery otherwise it gets confusing.

Reply:
Celeryworkers exit before unit tests run, so most of the tests are ran with a user in block execution mode. The jira_import_and_pusing_api tests have a user that user uses block execution, but the dojo_api_test_case does not. The functions I am using are in the dojo_api_test_case so the add_epic function will run async. With celery not active, the function never runs. Maybe I could try putting these functions in the same file as the unit test and see what happens.

Reply:
So all the tests here are run through the API under a user with a token. This function is ran unauthenticated though. So there is no way to specify it be ran under block execution. Having a sync and async version of the function is the only way. This is not confusing in the slightest though. The function is ran through celery in every single instance except this unit test. It's no different than the old way of doing things with functions set up like utils/tasks.

Reply:
That's whay I meant by that the engagement should not be created from the unittest. The unit test should call the rest api or the ui api with a post to create the engagement. It should assert that the epic is created if the epic mapping is enabled. Another test should assert an epic is not created if the epic mapping is not enabled.

Reply:
Epic mapping and epic creation are not a package deal.